packages/riverpod_lint/README.md

```diff
@@ -531,6 +532,61 @@ class Example extends Anything {
 }
 ```
 
+### avoid_ref_inside_state_dispose
+
+Avoid using `Ref` in the dispose method.
+
+**Good**:
+
+```dart
+class MyWidget extends ConsumerStatefulWidget {
+  const MyWidget({super.key});
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _MyWidgetState();
+}
+
+class _MyWidgetState extends ConsumerState<MyWidget> {
+ 
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () async {
+        ref.read(provider).doSomething();
+        return true;
+      },
+      child: Container(),
+    );
+  }
+}
+```
```

Comment:
This doesn't seem to add too much value. Let's remove it

````suggestion
**Good**:

```dart
class MyWidget extends ConsumerStatefulWidget {
  const MyWidget({super.key});

  @override
  ConsumerState<ConsumerStatefulWidget> createState() => _MyWidgetState();
}

class _MyWidgetState extends ConsumerState<MyWidget> {
 
  @override
  Widget build(BuildContext context) {
    return WillPopScope(
      onWillPop: () async {
        ref.read(provider).doSomething();
        return true;
      },
      child: Container(),
    );
  }
}
```
````